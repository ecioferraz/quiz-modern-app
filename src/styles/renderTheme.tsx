import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export default function renderTheme(children: ReactNode): RenderResult {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
}
