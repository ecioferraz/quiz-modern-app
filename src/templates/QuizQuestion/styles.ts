import styled, { css } from 'styled-components';

export const Container = styled.form`
  ${({ theme }) => css`
    h1 {
      font-weight: ${theme.font.weight.semiBold};
      margin-right: ${theme.spacings.huge};
    }

    img {
      margin-bottom: ${theme.spacings.xsmall};
      width: 100%;
    }
  `}
`;

export const ButtonsContainer = styled.section`
  ${({ theme }) => css`
    align-items: center;
    display: flex;
    justify-content: space-between;
    margin-top: 5.8rem;

    button {
      font-size: 1.4rem;
      font-weight: ${theme.font.weight.semiBold};
      width: 48%;

      :first-child {
        background: #FAFAFA;
        border: 1px solid #21BDCA;
        color: #21BDCA;
      }

      :last-child {
        background: #21BDCA;
        color: #FFF;
      }
    }
  `}
`;
