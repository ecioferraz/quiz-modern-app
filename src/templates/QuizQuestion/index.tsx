import Button from '../../components/Button';
import RadioInput from '../../components/RadioInput';
import RadioInputProps from '../../components/RadioInput/RadioInput.props';
import TextCard from '../../components/TextCard';
import * as Styled from './styles';

export default function QuizQuestion() {
  const questionOptions: RadioInputProps[] = [
    {
      answer: 'correct',
      iconSrc: 'assets/icons/circle.svg',
      option: 'Football',
    },
    {
      iconSrc: 'assets/icons/circle.svg',
      option: 'Basketball',
    },
    {
      answer: 'wrong',
      iconSrc: 'assets/icons/x.svg',
      option: 'Volleyball',
    },
    {
      iconSrc: 'assets/icons/circle.svg',
      option: 'Shuttlecock',
    },
  ];

  return (
    <Styled.Container aria-label="question">
      <TextCard>The image below is a field image used for?</TextCard>
      <img src="assets/images/question-image.png" />
      {questionOptions.map(({ iconSrc, answer, option }) => (
        <RadioInput
          answer={answer}
          iconSrc={iconSrc}
          key={option}
          option={option}
        />
      ))}
      <Styled.ButtonsContainer>
        <Button>Previous</Button>
        <Button>Next</Button>
      </Styled.ButtonsContainer>
    </Styled.Container>
  );
}
