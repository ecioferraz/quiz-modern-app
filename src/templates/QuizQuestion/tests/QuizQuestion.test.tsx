import renderTheme from '../../../styles/renderTheme';
import { screen } from '@testing-library/react';
import QuizQuestion from '..';

describe('<QuizQuestion />', () => {
  it('should render <QuizQuestion />', () => {
    renderTheme(<QuizQuestion />);
    const quizQuestion = screen.getByRole('form', { name: 'question' });

    expect(quizQuestion).toMatchSnapshot();
  });
});
