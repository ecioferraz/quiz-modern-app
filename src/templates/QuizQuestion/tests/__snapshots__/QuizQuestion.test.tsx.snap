// Vitest Snapshot v1

exports[`<QuizQuestion /> > should render <QuizQuestion /> 1`] = `
.c7 {
  border: none;
  border-radius: 10px;
  font-size: 1.4rem;
  font-weight: 600;
  height: 4.8rem;
  font-weight: 600;
  width: 100%;
}

.c1 {
  font-size: 1.6rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #219653;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
  margin-top: 1.6rem;
  padding: 1.6rem;
}

.c2 img {
  width: 2.4rem;
}

.c2 p {
  font-size: 1.4rem;
  font-weight: 500;
  margin: 0;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
  margin-top: 1.6rem;
  padding: 1.6rem;
}

.c4 img {
  width: 2.4rem;
}

.c4 p {
  font-size: 1.4rem;
  font-weight: 500;
  margin: 0;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #F6E5E5;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
  margin-top: 1.6rem;
  padding: 1.6rem;
}

.c5 img {
  width: 2.4rem;
}

.c5 p {
  font-size: 1.4rem;
  font-weight: 500;
  margin: 0;
}

.c0 h1 {
  font-weight: 600;
  margin-right: 5.6rem;
}

.c0 img {
  margin-bottom: 0.8rem;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 5.8rem;
}

.c6 button {
  font-size: 1.4rem;
  font-weight: 600;
  width: 48%;
}

.c6 button:first-child {
  background: #FAFAFA;
  border: 1px solid #21BDCA;
  color: #21BDCA;
}

.c6 button:last-child {
  background: #21BDCA;
  color: #FFF;
}

<form
  aria-label="question"
  class="c0"
>
  <h1
    class="c1"
  >
    The image below is a field image used for?
  </h1>
  <img
    src="assets/images/question-image.png"
  />
  <section
    class="c2"
  >
    <section
      class="c3"
    >
      <img
        alt="Radio option"
        src="assets/icons/circle.svg"
      />
    </section>
    <p
      class="c1"
    >
      Football
    </p>
  </section>
  <section
    class="c4"
  >
    <section
      class="c3"
    >
      <img
        alt="Radio option"
        src="assets/icons/circle.svg"
      />
    </section>
    <p
      class="c1"
    >
      Basketball
    </p>
  </section>
  <section
    class="c5"
  >
    <section
      class="c3"
    >
      <img
        alt="Radio option"
        src="assets/icons/x.svg"
      />
    </section>
    <p
      class="c1"
    >
      Volleyball
    </p>
  </section>
  <section
    class="c4"
  >
    <section
      class="c3"
    >
      <img
        alt="Radio option"
        src="assets/icons/circle.svg"
      />
    </section>
    <p
      class="c1"
    >
      Shuttlecock
    </p>
  </section>
  <section
    class="c6"
  >
    <button
      class="c7"
    >
      Previous
    </button>
    <button
      class="c7"
    >
      Next
    </button>
  </section>
</form>
`;
